lib/clouds/clouds/vsphere.rb

```diff
@@ -59,9 +76,105 @@ def requires_network_config?
       true
     end
 
+    def write_error(message)
+      logger.error(message) #STDERR.puts message
+    end
+
+    def write_debug(message)
+      logger.debug(message) #STDERR.puts message
+    end
+
+    def write_output(message)
+      logger.info(message) #STDOUT.puts message
+    end
+
+    def run_command(cmd)
+      output = `#{cmd}`
+      return [output, $?.exitstatus]
+    end
+
+    def query_data(type)
```

Comment:
Yes there was no advantage to having it in a separate file. It made the code less clear. I also couldn't write spec tests for this when it existed in right_link_package because I was running into gem dependency hell as it requires right_agent but its incompatible with other gems in right_link_package.